blueflood-core/src/main/java/com/rackspacecloud/blueflood/service/Configuration.java

```diff
@@ -17,25 +17,41 @@
 package com.rackspacecloud.blueflood.service;
 
 import com.google.common.collect.Lists;
+import com.rackspacecloud.blueflood.cache.ConfigTtlProvider;
 
 import java.io.IOException;
 import java.net.URL;
 import java.util.*;
 
 public class Configuration {
-    private static final Properties defaultProps = new Properties();
-    private static Properties props;
-    private static final Configuration INSTANCE = new Configuration();
+    private final Properties defaultProps;
+    private final Properties props;
+    private static final Configuration INSTANCE = new Configuration(true);
 
     public static Configuration getInstance() {
         return INSTANCE;
     }
 
-    private Configuration() {
-        try {
-            init();
-        } catch (IOException ex) {
-            throw new RuntimeException(ex);
+    /**
+     * Create a new instance of {@link Configuration} for testing.
+     */
+    public static Configuration getTestInstance(Properties defaultProps) {
+        Configuration config = new Configuration(false);
+        for (Map.Entry<Object, Object> entry : defaultProps.entrySet()) {
+            config.defaultProps.put(entry.getKey(), entry.getValue());
+        }
+        return config;
+    }
+
+    private Configuration(boolean shouldInitialize) {
```

Comment:
To be honest, I'm not exactly thrilled on the addition of parameter ```shouldInitialize``` to the constructor. It creates a different behavior of the class that skips the ```init()```, for the sake of testing. However, I don't have a better suggestion other than to mock Configuration class for testing. This has been done in some places in blueflood.